src/dlgPackageExporter.cpp

```diff
@@ -128,87 +316,135 @@ void dlgPackageExporter::recurseTree(QTreeWidgetItem* pItem, QList<QTreeWidgetIt
     }
 }
 
+bool dlgPackageExporter::writeFileToZip(const QString& archiveFileName, const QString& fileSystemFileName, zip* archive)
+{
+    struct zip_source* s = zip_source_file(archive, fileSystemFileName.toStdString().c_str(), 0, -1);
+    if (s == nullptr) {
+        displayResultMessage(tr("Failed to source file \"%1\" to place into module (archive) file, error is: \"%3\".")
+                             .arg(fileSystemFileName, QString::fromUtf8(zip_strerror(archive))), false);
+        return false;
+    }
+
+#if defined(LIBZIP_VERSION_MAJOR) && (LIBZIP_VERSION_MAJOR >= 1)
```

Comment:
Is this intended to be like this?

Reply:
Yeah, I was getting paranoid about the behaviour if `LIBZIP_VERSION_MAJOR` was NOT defined - normal short-cut rules mean (_I think_) that the second test will not be attempted if the first fails {like mentioned [here](https://stackoverflow.com/q/24037595) on Stack Overflow}. OTOH if it is not defined then the whole blooming `libzip` library is missing anyhow!

Reply:
... so the first test is pointless as `LIBZIP_VERSION_MAJOR` will appear as the value `0` when the expression is evaluated and it is not defined.  Will take the `defined(`...`)` tests out after all.